sdk/inc/azure/core/az_span.h

```diff
@@ -76,17 +76,24 @@ AZ_NODISCARD az_span az_span_create(uint8_t* ptr, int32_t size);
 #endif // AZ_NO_PRECONDITION_CHECKING
 
 /**
- * @brief An empty #az_span.
+ * @brief An empty #az_span literal.
```

Comment:
(adding the comment again - not sure why it's not listed).

I don't think we should add another public API for AZ_SPAN_EMPTY. We should instead use the existing API.

Even if we do decide to add the new API, please separate the PR / discussion.